src/Lib.P2P.Tests/Transports/TcpTest.cs

```diff
@@ -0,0 +1,232 @@
+            {
+                var _ = tcp.ConnectAsync("/tcp/32700").Result;
+            });
+        }
+
+        [TestMethod]
+        public void Connect_Unknown_Address()
+        {
+            var tcp = new Tcp();
+            ExceptionAssert.Throws<SocketException>(() =>
+            {
+                var _ = tcp.ConnectAsync("/ip4/127.0.10.10/tcp/32700").Result;
+            });
+        }
+
+        [TestMethod]
+        public void Connect_Cancelled()
+        {
+            var tcp = new Tcp();
+            var cs = new CancellationTokenSource();
+            cs.Cancel();
+            ExceptionAssert.Throws<OperationCanceledException>(() =>
+            {
+                var _ = tcp.ConnectAsync("/ip4/127.0.10.10/tcp/32700", cs.Token).Result;
+            });
+        }
+
+        [TestMethod]
+        [Ignore("Inconsistent results")]
+        public async Task TimeProtocol()
+        {
+            var cs = new CancellationTokenSource(TimeSpan.FromSeconds(30));
+            var server = await new MultiAddress("/dns4/time.nist.gov/tcp/37").ResolveAsync(cs.Token);
+            var data = new byte[4];
+
+            var tcp = new Tcp();
+            await using (var time = await tcp.ConnectAsync(server[0], cs.Token))
+            {
+                var n = await time.ReadAsync(data, 0, data.Length, cs.Token);
+                Assert.AreEqual(4, n); // sometimes zero!
+            }
+        }
+
+        [TestMethod]
+        public void Listen_Then_Cancel()
+        {
+            var tcp = new Tcp();
+            var cs = new CancellationTokenSource();
+
+            void Handler(Stream stream, MultiAddress local, MultiAddress remote)
+            {
+                Assert.Fail("handler should not be called");
+            }
+            
+            var listenerAddress = tcp.Listen("/ip4/127.0.0.1", Handler, cs.Token);
+            Assert.IsTrue(listenerAddress.Protocols.Any(p => p.Name == "tcp"));
+            cs.Cancel();
+        }
+
+        [TestMethod]
+        public async Task Listen()
+        {
+            var tcp = new Tcp();
+            var cs = new CancellationTokenSource(TimeSpan.FromSeconds(30));
+            var connected = false;
+            MultiAddress listenerAddress = null;
+
+            void Handler(Stream stream, MultiAddress local, MultiAddress remote)
+            {
+                Assert.IsNotNull(stream);
+                Assert.AreEqual(listenerAddress, local);
+                Assert.IsNotNull(remote);
+                Assert.AreNotEqual(local, remote);
+                connected = true;
+            }
+
+            try
+            {
+                listenerAddress = tcp.Listen("/ip4/127.0.0.1", Handler, cs.Token);
+                Assert.IsTrue(listenerAddress.Protocols.Any(p => p.Name == "tcp"));
+                await using (var stream = await tcp.ConnectAsync(listenerAddress, cs.Token))
+                {
+                    await Task.Delay(50, cs.Token);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4979568)